File: VB/WindowsApplication1/Helper/UnboundDataForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsApplication1.Helper
	Partial Public Class UnboundDataForm
		Inherits Form
		Private _Helper As MyUnboundGridHelper
		Public Property Helper() As MyUnboundGridHelper
			Get
				Return _Helper
			End Get
			Set(ByVal value As MyUnboundGridHelper)
				_Helper = value
				HelperChanged()
			End Set
		End Property

		Public Sub New(ByVal helper As MyUnboundGridHelper)
			InitializeComponent()
			Me.Helper = helper
		End Sub

		Private Sub HelperChanged()
			propertyGrid1.SelectedObject = Helper.GridData
			dataGridView1.DataSource = Helper.GridData
			dataNavigator1.DataSource = Helper.GridData
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/Helper/UnboundDataForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication1.Helper
{
    public partial class UnboundDataForm : Form
    {
        private MyUnboundGridHelper _Helper;
        public MyUnboundGridHelper Helper
        {
            get { return _Helper; }
            set { _Helper = value; HelperChanged(); }
        }

        public UnboundDataForm(MyUnboundGridHelper helper)
        {
            InitializeComponent();
            Helper = helper;
        }

        private void HelperChanged()
        {
            propertyGrid1.SelectedObject = Helper.GridData;
            dataGridView1.DataSource = Helper.GridData;
            dataNavigator1.DataSource = Helper.GridData;
        }
    }
}